_P248_TempHumidity_AHT1x.ino

```diff
@@ -0,0 +1,339 @@
+
+  if (humidity < 0)   return 0;
+  if (humidity > 100) return 100;
+  
+  return humidity;
+}
+
+
+
+
+
+/**************************************************************************/
+/*
+    readStatusByte()
+    Read status byte from sensor over I2C
+*/
+/**************************************************************************/
+unsigned char AHT1XReadStatus(uint8_t address) {
+  uint8_t result = 0;
+  // Need for AHT20?
+  //Wire.beginTransmission(aht1x_address);
+  //Wire.write(0x71);
+  //if (Wire.endTransmission() != 0) return false;
+  Wire.requestFrom(address, (uint8_t) 1);
+  result = Wire.read();
+  return result;
+}
+
+
+
+uint8_t AHT10_init(uint8_t address)
+{
+  Wire.beginTransmission(address);
+
+  Wire.write(AHTResetCmd);
+  Wire.endTransmission();
+  delay(AHT10_SOFT_RESET_DELAY);
+
+  Wire.beginTransmission(address);
+  Wire.write(AHTSetCalCmd, 3);
+  if (Wire.endTransmission() != 0)
+    return false;
+  delay(AHT10_MEASURMENT_DELAY);
+  if(AHT1XReadStatus(address) & 0x08) // Sensor calibrated?
+    return true;
+  return false;
+  
+  //while (AHT10_getStatus() & AHTX0_STATUS_BUSY) {
+  //  delay(10);
+  //}
+  //if (!(AHT10_getStatus() & AHTX0_STATUS_CALIBRATED)) {
+  //  return false;
+  //}
+
+  //if (Wire.endTransmission(true) != 0) return AHT10_ERROR;             //safety check, make sure transmission complete
+
+  //return 0x00;
+
+}
+
+
+
+//==============================================
+// PLUGIN
+// =============================================
+
+boolean Plugin_248(byte function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+    {
+      Device[++deviceCount].Number = PLUGIN_ID_248;
+      Device[deviceCount].Type = DEVICE_TYPE_I2C;
+      Device[deviceCount].VType = Sensor_VType::SENSOR_TYPE_TEMP_HUM;
+      Device[deviceCount].Ports = 0;
+      Device[deviceCount].PullUpOption = false;
+      Device[deviceCount].InverseLogicOption = false;
+      Device[deviceCount].FormulaOption = true;
+      Device[deviceCount].ValueCount = 2;
+      Device[deviceCount].SendDataOption = true;
+      Device[deviceCount].TimerOption = true;
+      Device[deviceCount].GlobalSyncOption = true;
+      break;
+    }
+
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_248);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_248));
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_248));
+      //strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[2], PSTR(PLUGIN_VALUENAME3_248));
+      //strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[3], PSTR(PLUGIN_VALUENAME4_248));
+    //  strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[4], PSTR(PLUGIN_VALUENAME5_248));
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SHOW_I2C_PARAMS:
+    {
+      byte choice = PCONFIG(0);
+
```

Comment:
Not really needed to allocate a variable for that, `PCONFIG(0)` can be used directly in the `addFormSelectorI2C()` call.

Reply:
ok.